octopoes/bits/missing_dmarc/missing_dmarc.py

```diff
@@ -1,19 +1,28 @@
-from typing import Dict, Iterator, List
+from typing import Dict, Iterator, List, Union
 
 import tldextract
 
 from octopoes.models import OOI
+from octopoes.models.ooi.dns.records import NXDOMAIN
 from octopoes.models.ooi.dns.zone import Hostname
 from octopoes.models.ooi.email_security import DMARCTXTRecord
 from octopoes.models.ooi.findings import Finding, KATFindingType
 
 
-def run(input_ooi: Hostname, additional_oois: List[DMARCTXTRecord], config: Dict[str, str]) -> Iterator[OOI]:
+def run(
+    input_ooi: Hostname, additional_oois: List[Union[DMARCTXTRecord, NXDOMAIN]], config: Dict[str, str]
+) -> Iterator[OOI]:
+    dmarc_records = [ooi for ooi in additional_oois if isinstance(ooi, DMARCTXTRecord)]
+    nxdomains = (ooi for ooi in additional_oois if isinstance(ooi, NXDOMAIN))
```

Comment:
we could return from the function in this loop immediately on the first NXDOMAIN instance.

Reply:
Yes that's why, as per ammars request, changed it from a list to a constructor

Reply:
> Yes that's why, as per ammars request, changed it from a list to a constructor

generator* :)